centreon/src/Utility/EnvironmentFileManager.php

```diff
@@ -0,0 +1,182 @@
+                    $this->add($key, $value);
+                }
+            }
+        } finally {
+            fclose($file);
+        }
+    }
+
+    /**
+     * @param string $key
+     * @param int|float|bool|string $value
+     */
+    public function add(string $key, int|float|bool|string $value): void
+    {
+        if ($value === 'true' || $value === 'false') {
+            $value = (bool) $value;
+        } elseif (preg_match('/^-?\d+\.\d+/', (string) $value)) {
+            $value = (float) $value;
+        } elseif (is_numeric($value)) {
+            if (str_starts_with($key, 'IS_') && $value === '1') {
+                $value = (bool) $value;
+            } else {
+                $value = (int) $value;
+            }
+        }
+        $this->variables[$key] = $value;
+    }
+
+    /**
+     * @param string $key
+     */
+    public function delete(string $key): void
+    {
+        unset($this->variables[$key]);
+    }
+
+    /**
+     * @return array<string, int|float|bool|string>
+     */
+    public function getAll(): array
+    {
+        return $this->variables;
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return int|float|bool|string|null
+     */
+    public function get(string $key): int|float|bool|string|null
+    {
+        return $this->variables[$key] ?? null;
+    }
+
+    /**
+     * @param string|null $environmentFile
+     *
+     * @throws \Exception
+     */
+    public function save(?string $environmentFile = null): void
+    {
+        $this->checkEnvironmentFileDefined($environmentFile);
+        $filePath = $this->environmentFilePath . $this->currentEnvironnementFile;
+        $file = fopen($filePath, 'w');
+        if (! $file) {
+            throw new \Exception(sprintf('Impossible to open file \'%s\'', $filePath));
+        }
+        try {
+            foreach ($this->variables as $key => $value) {
+                if (is_bool($value)) {
+                    $value = $value ? 1 : 0;
+                }
+                fwrite($file, sprintf("%s=%s\n", $key, $value));
+            }
+        } finally {
+            fclose($file);
+        }
+
+        $variables = var_export($this->variables, true);
+        $content = <<<CONTENT
+            <?php
+
+            // This file was generated by Centreon
+
+            return {$variables};
+            CONTENT;
+        file_put_contents($filePath . '.local.php', $content);
+    }
+
+    /**
+     * @param string|null $environmentFile
+     *
+     * @throws \Exception
+     */
+    private function checkEnvironmentFileDefined(?string $environmentFile = null): void
+    {
+        $environmentFile ??= $this->currentEnvironnementFile;
+        if ($environmentFile === null) {
+            throw new \Exception('No environment file defined');
+        }
+    }
+
+    /**
+     * Add the directory separator at the end of the path if it does not exist.
+     *
+     * /dir1/dir2 => /dir1/dir2/
+     *
+     * @param string $path
```

Comment:
Maybe use the `non-empty-string` phpdoc type because the function does not manage this case ?
```suggestion
     * @param non-empty-string $path
```